src/main/java/org/fxmisc/flowless/TargetPosition.java

```diff
@@ -30,6 +30,9 @@ public TargetPosition transformByChange(
         if(itemIndex >= pos + removedSize) {
             // change before the target item, just update item index
             return new StartOffStart(itemIndex - removedSize + addedSize, offsetFromStart);
+        } else if (pos <= itemIndex && itemIndex <= pos + removedSize) {
+            // deletion occurred before viewport and partially into it, so update the item index
+            return new StartOffStart(itemIndex - removedSize + addedSize, offsetFromStart);
         } else if(itemIndex >= pos) {
             // target item deleted, show the first inserted at the target offset
             return new StartOffStart(pos, offsetFromStart);
```

Comment:
This doesn't make sense to me. First of all, the condition ` && itemIndex <= pos + removedSize` is superfluous: the test above failed, so by this time we know that `itemIndex < pos + removedSize`. So it is effectively overriding the next `else if` branch.